src/txn/txn_rollback_to_stable.c

```diff
@@ -386,8 +386,20 @@ __rollback_ondisk_fixup_key(WT_SESSION_IMPL *session, WT_REF *ref, WT_ROW *rip,
       __wt_key_string(session, key->data, key->size, S2BT(session)->key_format, key_string));
 
     WT_ERR(__wt_scr_alloc(session, 0, &full_value));
-    WT_ERR(__wt_page_cell_data_ref(session, page, unpack, full_value));
-    WT_ERR(__wt_buf_set(session, full_value, full_value->data, full_value->size));
+    WT_ERR_ERROR_OK(
+      __wt_page_cell_data_ref_kv(session, page, unpack, full_value), WT_RESTART, true);
+    /*
+     * We can read overflow removed value if checkpoint has run before rollback to stable. In this
+     * case, we have already appended the on page value to the update chain. At this point, we have
+     * visited the update chain and decided the value is not stable. In addition, checkpoint must
+     * have moved this value to the history store as a full value. Therefore, we can safely ignore
+     * the on page value if it is overflow removed.
+     */
+    if (ret == 0)
+        WT_ERR(__wt_buf_set(session, full_value, full_value->data, full_value->size));
+    else
```

Comment:
Inside `__wt_page_cell_data_ref_kv `, we call `__cell_data_ref` that might return something else than `0` or `WT_RESTART`. Should we do instead:
```suggestion
    else if(ret == WT_RESTART) ret = 0;
    WT_ERR(ret);
```

Reply:
WT_ERR_ERROR_OK macro is already handling this

Reply:
That's right, my bad, thanks.